packages/backend/src/services/SlackService/SlackService.ts

```diff
@@ -0,0 +1,530 @@
+                        accessory: {
+                            type: 'button',
+                            text: {
+                                type: 'plain_text',
+                                text: 'Open in Lightdash',
+                                emoji: true,
+                            },
+                            url,
+                            action_id: 'button-action',
+                        },
+                    },
+                    {
+                        type: 'image',
+                        image_url: imageUrl,
+                        alt_text: chart.name,
+                    },
+                ],
+            },
+        };
+    }
+
+    private async unfurlDashboard(
+        url: string,
+        sharedUrl: string,
+        imageUrl: string,
+    ): Promise<any> {
+        const [projectUuid, dashboardUuid] =
+            (await sharedUrl.match(uuidRegex)) || [];
+
+        const dashboard = await this.dashboardModel.getById(dashboardUuid);
+
+        // https://api.slack.com/reference/block-kit/blocks
+        // https://app.slack.com/block-kit-builder/T0163M87MB9#%7B%22blocks%22:%5B%5D%7D
+        const unfurls = {
+            [url]: {
+                blocks: [
+                    {
+                        type: 'header',
+                        text: {
+                            type: 'plain_text',
+                            text: dashboard.name,
+                        },
+                    },
+                    {
+                        type: 'section',
+                        text: {
+                            type: 'mrkdwn',
+                            text: `${dashboard.description || '-'}`,
+                        },
+                        accessory: {
+                            type: 'button',
+                            text: {
+                                type: 'plain_text',
+                                text: 'Open in Lightdash',
+                                emoji: true,
+                            },
+                            url,
+                            action_id: 'button-action',
+                        },
+                    },
+                    {
+                        type: 'image',
+                        image_url: imageUrl,
+                        alt_text: dashboard.name,
+                    },
+                ],
+            },
+        };
+        return unfurls;
+    }
+
+    private async getSharedUrl(linkUrl: string): Promise<string> {
+        const [shareId] = linkUrl.match(nanoidRegex) || [];
+        const shareUrl = await this.shareModel.getSharedUrl(shareId);
+
+        const fullUrl = `${this.lightdashConfig.siteUrl}${shareUrl.path}${shareUrl.params}`;
+        Logger.debug(`Shared url ${shareId}: ${fullUrl}`);
+
+        return fullUrl;
+    }
+
+    private async parseUrl(linkUrl: string): Promise<{
+        isValid: boolean;
+        lightdashPage?: LightdashPage;
+        url: string;
+    }> {
```

Comment:
Type this response, and include projectUuid? dashboardUuid? chartUuid? and explore model name, so we do all the parsing here, and we don't need to pass the URL to unfurlPage

Reply:
Created a new issue for this ^